### PR TITLE
when we delete sites it will also destroy tasks

### PR DESCRIPTION
**Pull request by ArtemRakov**

### CODE REVIEW

File: apps/web/controllers/sites/destroy.rb [hunk 1]
```diff
@@ -5,11 +5,17 @@ class Destroy
...
+          @site = repo.find(params[:id])
+          repo.delete_with_dependencies(@site.id)
 
           redirect_to routes.sites_path
         end
+
+        private
+
+        def repo
+          @repo ||= SiteRepository.new
```
Comment (Mehonoshin) [inline apps/web/controllers/sites/destroy.rb]: inject with dry-system please

File: lib/omniscrapper_hq/repositories/site_repository.rb [hunk 2]
```diff
@@ -6,4 +6,17 @@ class SiteRepository < Hanami::Repository
...
+  def delete_with_dependencies(id)
+    delete_dependent_tasks(id)
+    delete(id)
+  end
+
+  def delete_dependent_tasks(id)
+    TaskRepository.new.root.where(site_id: id).delete
+  end
+
+  def find_with_tasks(id)
```
Comment (Mehonoshin) [inline lib/omniscrapper_hq/repositories/site_repository.rb]: where is this method used?

File: apps/web/controllers/sites/destroy.rb (hunk 1 above)
Comment (ArtemRakov) [reply apps/web/controllers/sites/destroy.rb]: Done

File: lib/omniscrapper_hq/repositories/site_repository.rb (hunk 2 above)
Comment (ArtemRakov) [reply lib/omniscrapper_hq/repositories/site_repository.rb]: It was added mostly for testing purposes. (removed)

File: lib/omniscrapper_hq/repositories/task_repository.rb [hunk 3]
```diff
@@ -9,6 +9,12 @@ class TaskRepository < Hanami::Repository
     belongs_to :site
   end
 
+  def delete_related_to(options)
+    root.where(options).each do |task|
+      delete_with_dependencies(task.id)
```
Comment (ArtemRakov) [inline lib/omniscrapper_hq/repositories/task_repository.rb]: n + 1 will occur here. Do you know how to solve this problem with rom rb ? :)

Comment (ArtemRakov) [reply lib/omniscrapper_hq/repositories/task_repository.rb]: or it is ok for now?

**Pull request merged**